framework/include/materials/MaterialBase.h

```diff
@@ -280,6 +284,18 @@ class MaterialBase : public MooseObject,
 template <typename T>
 MaterialProperty<T> &
 MaterialBase::declareProperty(const std::string & prop_name)
+{
+  // Check if the supplied parameter is a valid input parameter key
+  std::string name = prop_name;
+  if (_pars.have_parameter<MaterialPropertyName>(prop_name))
+    name = _pars.get<MaterialPropertyName>(prop_name);
+
+  return declarePropertyByName<T>(name);
+}
+
```

Comment:
```suggestion
MaterialBase::declareProperty(const std::string & name)
{
  // Check if the supplied parameter is a valid input parameter key
  std::string prop_name = name;
  if (_pars.have_parameter<MaterialPropertyName>(name))
    prop_name = _pars.get<MaterialPropertyName>(name);

  return declarePropertyByName<T>(prop_name);
}

```

Reply:
I thought this is completely logical. The input parameter is either a property name or an input parameter name, so we use the least specific name here.